runtime/src/main/java/org/corfudb/runtime/view/workflows/AddNode.java

```diff
@@ -55,4 +68,25 @@ protected boolean verifyRequest(@NonNull Layout layout) {
     public String toString() {
         return this.getClass().getSimpleName() + " " + nodeForWorkflow;
     }
+
+    @Override
+    protected Optional<ManagementClient> getOrchestrator() {
+        runtime.invalidateLayout();
+        Layout layout = new Layout(runtime.getLayoutView().getLayout());
+        BaseClient baseClient = runtime.getLayoutView().getRuntimeLayout(layout).getBaseClient(nodeForWorkflow);
```

Comment:
For AddNode workflow If the node to be added is selected as orchestrator, how does it add itself to the cluster? It's not part of the cluster yet...right?

Reply:
Its bootstrapped with a management layout before this workflowss begins